### PR TITLE
chore(macros/CSSRef): extend + update sidebar

### PR DESCRIPTION
**Pull request by glmvc**
## Summary

adapting things in the sidebar to fit the changes in pr #5828 and to be more consistent like the HTML sidebar e.g. (pr #6050)

### Problem

- For the [CSS building blocks module](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks) the assessments are missing in the sidebar.
- In the [CSS layout module](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout) "Assessment: " is missing before the assessment title in the sidebar.
- In the [CSS first steps module](https://developer.mozilla.org/en-US/docs/Learn/CSS/First_steps) the assessment got a new title (see this https://github.com/mdn/content/pull/13781)
- Sometimes the capitalization was different from the article titles themselves

### Solution

- updated the HTML template by adding the missing assessments
- added "Assessment: " before the assessment title
- adapted the string identifiers and translations for the new assessment title
- adapted the capitalization to fit the article titles
- adapted string identifiers to fit translations better

---

### Before

image

### After

image

---

## How did you test this change?

- checked if the sidebar is the same as the one in the learning area
- checked for flaws
- ckecked if links/urls are working and redirecting to the correct page

### CODE REVIEW

**Comment by glmvc** [comment]
#### I tested all additional changes (new commits) by setting up yarn and checking the sidebar (textually and if the links work) ✔️

**Comment by glmvc** [comment]
Like mentioned in pr #5828, what about missing translations here?

Translations present:
- English (US) of course
- German
- Japanese
- Portuguese (Brazilian)
- Russian
- Chinese (both simplified and traditional)

Languages missing (but offered on MDN):
- French (Note [Issue #5135](https://github.com/mdn/translated-content/issues/5135))
- Spanish
- Polish
- Korean

**Comment by glmvc** [comment]
What I also noticed is that this file is generally structured differently than, for example, LearnSidebar.ejs or HTMLSidebar.ejs.
- Is there a specific reason for this?
- Is this still the old version, or even a newer one?
- Shouldn't these files be consistent in structure and function?

**Comment by glmvc** [comment]
I've also added a link to "Tutorials:" in the sidebar to point to Learn/CSS- I thought it's a good idea, because the other "Headings" in the sidebar are also pointing to somewhere where it fits.

### Suggestion: 
Shouldn't "Tutorials" be renamed to "Learn [CSS in this case]"?
As far as I have seen on mdn, this term is never used. Instead guides, learn ..., articles, modules etc. are used.

**Comment by glmvc** [comment]
Hi @caugner, i've done a merge.
Also, i have here almost the same suggestions / questions as in the _already_ merged pr #6050.

Should I rather make a separate pr with these suggestions?

**Comment by glmvc** [comment]
@wbamberg @caugner hi, friendly reminder: this pr is still waiting for your review :) thanks and have a nice day

File: kumascript/macros/CSSRef.ejs [hunk 1]
```diff
@@ -148,6 +152,54 @@ const text = mdn.localStringMap({
       'Box-shadow_generator': 'Box shadow generator',
       'Border-image_generator' : 'Border image generator',
   },
+  'de': {
```
Comment (caugner) [inline kumascript/macros/CSSRef.ejs:155]: German is now retired, so this key needs to be dropped.

File: kumascript/macros/CSSRef.ejs [hunk 2]
```diff
@@ -296,14 +348,15 @@ const text = mdn.localStringMap({
...
+    'Tutorials': 'Tutorials',
+    'CSS_basics': 'CSS basics',
       'CSS_first_steps': 'CSS の第一歩',
         'CSS_first_steps_overview': 'CSS の第一歩の概要',
         'What_is_CSS':  'CSS とは何か',
         'Getting_started_with_CSS': 'CSS 入門',
         'How_CSS_is_structured': 'CSS の全体像',
         'How_CSS_works': 'CSS の働き',
-        'Using_your_new_knowledge': '新しい知識を使う',
+        'Assessment_Styling_a_biography_page': 'Assessment: Styling a biography page',
```
Comment (caugner) [inline kumascript/macros/CSSRef.ejs:359]: @mdn/yari-content-ja Could you please review these changes?

File: kumascript/macros/CSSRef.ejs [hunk 3]
```diff
@@ -357,43 +414,47 @@ const text = mdn.localStringMap({
...
+        'Multiple-column_layout': 'Multiple-column layout',
         'Responsive_design': 'Responsive design',
-        'Media_queries': 'Beginner\'s guide to media queries',
-        'Legacy_Layout_Methods': 'Legacy Layout Methods',
-        'Supporting_Older_Browsers': 'Supporting Older Browsers',
-        'Fundamental_Layout_Comprehension': 'Fundamental Layout Comprehension',
+        'Beginners_guide_to_media_queries': 'Beginner\'s guide to media queries',
+        'Legacy_layout_methods': 'Legacy layout methods',
+        'Supporting_older_browsers': 'Supporting older browsers',
+        'Assessment_Fundamental_Layout_Comprehension': 'Assessment: Fundamental Layout Comprehension',
```
Comment (caugner) [inline kumascript/macros/CSSRef.ejs:446]: @mdn/yari-content-pt-br Could you please review these changes?

File: kumascript/macros/CSSRef.ejs [hunk 4]
```diff
@@ -357,43 +414,47 @@ const text = mdn.localStringMap({
...
+    'Tutorials': 'Уроки',
+    'CSS_basics': 'Основы CSS',
       'CSS_first_steps': 'Введение в CSS',
         'CSS_first_steps_overview': 'Введение в CSS',
         'What_is_CSS':  'Что такое CSS?',
         'Getting_started_with_CSS': 'Начало работы с CSS',
         'How_CSS_is_structured': 'Как структурирован CSS',
         'How_CSS_works': 'Как работает CSS',
-        'Using_your_new_knowledge': 'Использование ваших новых знаний',
+        'Assessment_Styling_a_biography_page': 'Assessment: Styling a biography page',
```
Comment (caugner) [inline kumascript/macros/CSSRef.ejs:457]: @mdn/yari-content-ru Could you please review this change?

File: kumascript/macros/CSSRef.ejs [hunk 5]
```diff
@@ -402,19 +463,22 @@ const text = mdn.localStringMap({
...
-        'Values_and_units': 'Значения свойств CSS',
+        'CSS_values_and_units': 'Значения свойств CSS',
         'Sizing_items_in_CSS': 'Изменение размеров в CSS',
         'Images_media_and_form_elements': 'Элементы изображений, медиа и форм',
         'Styling_tables': 'Стилизация таблиц',
         'Debugging_CSS': 'Отладка CSS',
         'Organizing_your_CSS': 'Организация CSS-кода',
+        'Assessment_Fundamental_CSS_comprehension': 'Assessment: Fundamental CSS comprehension',
+        'Assessment_Creating_fancy_letterheaded_paper': 'Assessment: Creating fancy letterheaded paper',
+        'Assessment_A_cool_looking_box': 'Assessment: A cool-looking box',
```
Comment (caugner) [inline kumascript/macros/CSSRef.ejs:474]: @mdn/yari-content-ru Could you please review these changes?

File: kumascript/macros/CSSRef.ejs [hunk 6]
```diff
@@ -549,19 +614,22 @@ const text = mdn.localStringMap({
...
-        'Values_and_units': '值与单位',
+        'CSS_values_and_units': 'CSS values and units',
         'Sizing_items_in_CSS': '在 CSS 中调整大小',
         'Images_media_and_form_elements': '图像、媒体和表单元素',
         'Styling_tables': '样式化表格',
         'Debugging_CSS': '调试 CSS',
         'Organizing_your_CSS': '组织你的 CSS',
+        'Assessment_Fundamental_CSS_comprehension': 'Assessment: Fundamental CSS comprehension',
+        'Assessment_Creating_fancy_letterheaded_paper': 'Assessment: Creating fancy letterheaded paper',
+        'Assessment_A_cool_looking_box': 'Assessment: A cool-looking box',
```
Comment (caugner) [inline kumascript/macros/CSSRef.ejs:625]: @mdn/yari-content-zh Could you please review these `zh-CN` changes?

File: kumascript/macros/CSSRef.ejs [hunk 7]
```diff
@@ -696,32 +765,35 @@ const text = mdn.localStringMap({
...
+        'Multiple-column_layout': 'Multiple-column layout',
         'Responsive_design': 'Responsive design',
-        'Media_queries': 'Beginner\'s guide to media queries',
-        'Legacy_Layout_Methods': 'Legacy Layout Methods',
-        'Supporting_Older_Browsers': 'Supporting Older Browsers',
-        'Fundamental_Layout_Comprehension': 'Fundamental Layout Comprehension',
+        'Beginners_guide_to_media_queries': 'Beginner\'s guide to media queries',
+        'Legacy_layout_methods': 'Legacy layout methods',
+        'Supporting_older_browsers': 'Supporting older browsers',
+        'Assessment_Fundamental_Layout_Comprehension': 'Assessment: Fundamental Layout Comprehension',
```
Comment (caugner) [inline kumascript/macros/CSSRef.ejs:796]: @mdn/yari-content-zh Could you please review these `zh-TW` changes?

**Review by wbamberg — COMMENTED** [review]
Thanks for your PR, @glmvc , and sorry to take so long to review it. I like all the fixes to page titles.

The changes at the top could use a bit more discussion. The current page:
Screen Shot 2022-08-31 at 12 12 44 PM

This PR:

Screen Shot 2022-08-31 at 12 12 32 PM

I guess you have taken the cue from the [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML) and [JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript) sidebars here, which:

- have "HTML"/"JavaScript" at the top
- calls Learn "Tutorials:"
- has the "HTML basics" before the HTML Learn tutorials

So I guess this makes our sidebars more consistent, which is a good thing! Overall though I think our sidebars are so inconsistent that it would take a complete project to overhaul them. It's hard to know how to assess piecemeal changes like this.

Personally:

- If we want to have a colon after "Tutorials" we should do the same for "Reference" and "Guides".
- The HTML/JS sidebars also have "References:" (plural) instead of "Reference".
- I'm not keen on the [CSS basics](http://localhost:3000/en-US/docs/Learn/Getting_started_with_the_web/CSS_basics) link at the start. I feel like this is part of a different tutorial and will make things quite fragmentary (I mean that if you follow the "Next" link from this page, you end up outside the CSS realm).

cc @dipikabh who has been thinking about some of these things.

**Comment by caugner** [comment]
FWIW I just realize that the CSSRef sidebar shows "Learn CSS" (which points to a different area) before the actual "Reference" links. That's confusing, because if you click on any of those links, you will get a totally different sidebar.

File: kumascript/macros/CSSRef.ejs (hunk 6 above)
Comment (yin1999) [reply kumascript/macros/CSSRef.ejs:625]: Let's translate it:

```suggestion
        'CSS_values_and_units': 'CSS 值和单位',
        'Sizing_items_in_CSS': '在 CSS 中调整大小',
        'Images_media_and_form_elements': '图像、媒体和表单元素',
        'Styling_tables': '样式化表格',
        'Debugging_CSS': '调试 CSS',
        'Organizing_your_CSS': '组织你的 CSS',
        'Assessment_Fundamental_CSS_comprehension': '测验：基本的 CSS 理解',
        'Assessment_Creating_fancy_letterheaded_paper': '测验：创建精美的信纸',
        'Assessment_A_cool_looking_box': '测验：一个漂亮的盒子',
```

File: kumascript/macros/CSSRef.ejs [hunk 8]
```diff
@@ -570,12 +638,12 @@ const text = mdn.localStringMap({
...
         'Floats': '浮动',
         'Positioning': '定位',
-        'Multiple-column_Layout': '多栏式布局',
+        'Multiple-column_layout': '多栏式布局',
         'Responsive_design': '响应式布局',
-        'Media_queries': '媒体查询',
-        'Legacy_Layout_Methods': '传统的布局方法',
-        'Supporting_Older_Browsers': '支持旧版浏览器',
-        'Fundamental_Layout_Comprehension': '测验：对布局基础知识的理解',
+        'Beginners_guide_to_media_queries': 'Beginner\'s guide to media queries',
```
Comment (yin1999) [inline kumascript/macros/CSSRef.ejs:643]:
```suggestion
        'Beginners_guide_to_media_queries': '媒体查询入门指南',
```

File: kumascript/macros/CSSRef.ejs [hunk 9]
```diff
@@ -533,14 +597,15 @@ const text = mdn.localStringMap({
...
+    'Tutorials': 'Tutorials',
+    'CSS_basics': 'CSS basics',
       'CSS_first_steps': 'CSS 第一步',
         'CSS_first_steps_overview': 'CSS 第一步概述',
         'What_is_CSS':  '什么是 CSS?',
         'Getting_started_with_CSS': 'CSS 入门',
         'How_CSS_is_structured': 'CSS 的结构是怎样的',
         'How_CSS_works': 'CSS 是如何工作的',
-        'Using_your_new_knowledge': '运用你的新知识',
+        'Assessment_Styling_a_biography_page': '作业：传记页的风格设计',
```
Comment (yin1999) [inline kumascript/macros/CSSRef.ejs:608]:
```suggestion
        'Assessment_Styling_a_biography_page': '测验：为传记页面添加样式',
```

File: kumascript/macros/CSSRef.ejs [hunk 10]
```diff
@@ -533,14 +597,15 @@ const text = mdn.localStringMap({
       'Border-image_generator' : 'Генератор Border-image',
   },
   'zh-CN': {
-    'Learn_CSS': '学习 CSS',
+    'Tutorials': 'Tutorials',
+    'CSS_basics': 'CSS basics',
```
Comment (yin1999) [inline kumascript/macros/CSSRef.ejs:601]:
```suggestion
    'Tutorials': '教程',
    'CSS_basics': 'CSS 基础',
```

File: kumascript/macros/CSSRef.ejs [hunk 11]
```diff
@@ -680,14 +748,15 @@ const text = mdn.localStringMap({
       'Border-image_generator' : '图片边框生成器',
   },
   'zh-TW': {
-    'Learn_CSS': 'Learn CSS',
+    'Tutorials': 'Tutorials',
+    'CSS_basics': 'CSS basics',
```
Comment (yin1999) [inline kumascript/macros/CSSRef.ejs:752]:
```suggestion
    'Tutorials': '教程',
    'CSS_basics': 'CSS 基礎',
```

File: kumascript/macros/CSSRef.ejs (hunk 7 above)
Comment (yin1999) [reply kumascript/macros/CSSRef.ejs:796]: /cc @irvin

**Review by yin1999 — COMMENTED** [review]
some translation for zh-CN and zh-TW

**Comment by wbamberg** [comment]
> FWIW I just realize that the CSSRef sidebar shows "Learn CSS" (which points to a different area) before the actual "Reference" links. That's confusing, because if you click on any of those links, you will get a totally different sidebar.

This is true, and happens all over the place (the other sidebars do a similar thing). But not clear how best to fix this without a thorough IA for the site.

**Review by mfuji09 — COMMENTED** [review]
I have reviewed the texts of `ja` and suggested some new translations.

File: kumascript/macros/CSSRef.ejs [hunk 12]
```diff
@@ -148,6 +152,54 @@ const text = mdn.localStringMap({
...
+        'Grids' : 'Grid',
+        'Floats': 'Float',
+        'Positioning': 'Positionierung',
+        'Multiple-column_layout': 'Multiple-column layout',
+        'Responsive_design': 'Responsive design',
+        'Beginners_guide_to_media_queries': 'Beginner\'s guide to media queries',
+        'Legacy_layout_methods': 'Legacy layout methods',
+        'Supporting_older_browsers': 'Supporting older browsers',
+        'Assessment_Fundamental_Layout_Comprehension': 'Assessment: Fundamental layout comprehension',
+  },
```
Comment (caugner) [inline kumascript/macros/CSSRef.ejs:202]: Remove German

```suggestion
```

**Comment by caugner** [comment]
> Personally:
> 
> * If we want to have a colon after "Tutorials" we should do the same for "Reference" and "Guides".
> * The HTML/JS sidebars also have "References:" (plural) instead of "Reference".
> * I'm not keen on the [CSS basics](http://localhost:3000/en-US/docs/Learn/Getting_started_with_the_web/CSS_basics) link at the start. I feel like this is part of a different tutorial and will make things quite fragmentary (I mean that if you follow the "Next" link from this page, you end up outside the CSS realm).

@wbamberg Would you be okay to have this PR merged as is (after removing the added colon)?

* The "Reference" item was already called "Reference" before, and the corresponding page is "CSS reference", so it seems correct.
* The "CSS basics" page is indeed part of the "Getting started with the web" section, with a different sidebar, but the [JavaScript section](http://localhost:3000/en-US/docs/Web/JavaScript/) has the same issue, except that the sidebar is less detailed, so "JavaScript basics", "JavaScript first steps" and "JavaScript building blocks" (despite having subpages) only have one menu item inside a "Complete beginners" submenu. 🤷

**Comment by wbamberg** [comment]
> @wbamberg Would you be okay to have this PR merged as is (after removing the added colon)?

TL:DR; yes, reluctantly :).

> * The "Reference" item was already called "Reference" before, and the corresponding page is "CSS reference", so it seems correct.

Sure. 

> * The "CSS basics" page is indeed part of the "Getting started with the web" section, with a different sidebar, but the [JavaScript section](https://developer.mozilla.org/en-US/docs/Web/JavaScript/) has the same issue, except that the sidebar is less detailed, so "JavaScript basics", "JavaScript first steps" and "JavaScript building blocks" (despite having subpages) only have one menu item inside a "Complete beginners" submenu: 🤷

I would still like to remove "CSS basics" here. I don't mind our reference docs linking to their corresponding Learn section (e.g. https://developer.mozilla.org/en-US/docs/Learn/CSS/First_steps) because you stay in CSS-land when you navigate around there. But I'm much less keen on them linking to pages under the https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web module , because that really feels like different territory.

I take your point that the JS sidebar does this, but (1) I would remove it from there too and (2) it doesn't seem as bad there because it's hidden a bit under a "Complete beginners" section, rather than here where it's out on its own.

Still I won't block this PR on it.

**Comment by caugner** [comment]
FWIW The HTMLRef sidebar also has an "HTML basics" item: https://developer.mozilla.org/en-US/docs/Web/HTML/Reference

Let's add it here to be consistent.

File: kumascript/macros/CSSRef.ejs [hunk 13]
```diff
@@ -696,32 +717,35 @@ const text = mdn.localStringMap({
...
+        'Assessment_Creating_fancy_letterheaded_paper': 'Assessment: Creating fancy letterheaded paper',
+        'Assessment_A_cool_looking_box': 'Assessment: A cool-looking box',
       'Styling_text': '樣式化文字',
         'Styling_text_overview': '樣式化文字概述',
         'Fundamental_text_and_font_styling': '基礎的文字與字型樣式化',
         'Styling_lists': '樣式化列表',
         'Styling_links': '樣式化連結',
         'Web_fonts': 'Web 字型',
-        'Assessment_typesetting_a_community_school_homepage': '親合度：設定社區大學首頁的版面',
+        'Assessment_Typesetting_a_community_school_homepage': '親合度：設定社區大學首頁的版面',
```
Comment (irvin) [inline kumascript/macros/CSSRef.ejs:735]:
```suggestion
        'Assessment_Typesetting_a_community_school_homepage': '練習：設定社區大學首頁的版面',
```

**Review by irvin — CHANGES_REQUESTED** [review]

File: kumascript/macros/CSSRef.ejs (hunk 13 above)
Comment (irvin) [reply kumascript/macros/CSSRef.ejs:735]: just take chance to fix this one, others were all fine

**Review by caugner — APPROVED** [review]

**Review by caugner — APPROVED** [review]

**Comment by glmvc** [comment]
@wbamberg , @caugner & @ translators :) 
thank you for your efforts and discussion.

i think the merge was a good idea (thx) since this PR has been pending for a while now, but you're right: greater consistency would definitely make sense - but of course needs more detailed reflections and discussions.
